### ISSUE TITLE
shift+command+w close current window on vscode does not trigger relayout

### ISSUE THREAD

**Issue by eret9616** [koekeishiya/yabai#2431]
In VSCode, pressing Shift + Cmd + w to close the current window doesn’t trigger a relayout

yabai:
yabai-v7.1.3

vscode:
code 1.88.1
e170252f762678dec6ca2cc69aba1570769a5d39
arm64

Reproduction steps:

	1.	Open iTerm terminal and place it on the left side, open VSCode and place it on the right side.
	2.	Focus on VSCode, then press Shift + Cmd + W to close the VSCode window.
	3.	The windows do not reallocate.

**Comment by penamimax** [comment]
I'm facing the same issue with pretty much every program. I.e., I can't name a single program where this doesn't happen.
Also on yabai-v7.1.3 on MacOS 15.0. 

Switching spaces back and forth on the specific display where the empty space remained seems to cause Yabai to relayout and fill the empty space again. 

It might be related to issue #2426 . Just that it seems way more widespread.

**Comment by Liquidmantis** [comment]
Oooh, turned out to be Contexts for me, too.  That sucks, but I don't use Contexts that heavily anyway.

**Comment by penamimax** [comment]
This is puzzling. It almost is as if Yabai lost the ability to detect windows being closed on my system.
I don't use Contexts, but Phoenix (https://github.com/kasper/phoenix) and unfortunately I use that extensively.

E.g., I even replaced skhd with phoenix since I was already using for various other tasks, like manual tiling when not using Yabai's bsp window arrangement.

I can confirm that disabling phoenix makes Yabai detect closed windows again. 

Issue #2324 describes why this happened.

So bottom line, it seems like we can only use one window arranging program at a time, even if the two are not actively arranging windows at the same time. That's unfortunate.

Since this is a MacOS based change, don't feel obligated to keep that issue opened on my behalf.

**Comment by eret9616** [comment]
Today I tried to disable the context, but I found that installing and using Amazon Q also causes this issue…

**Comment by koekeishiya** [comment]
Try the latest commit from master; I've attempted a workaround, but this is either a system-issue caused by an Apple bug, or a bug triggered by the other third-party software somehow. I don't know how contexts or amazon q works.

**Comment by koekeishiya** [comment]
Just did a release after testing the changes.

**Comment by john-persson-polestar** [comment]
I just updated to v7.1.4, and this is still a problem for me on Sequoia.
https://github.com/user-attachments/assets/79d8388c-a991-4e05-ab9a-98b6ec4daa0d

**Comment by johnnyBira** [comment]
I just updated to v7.1.4, and this is still a problem for me on Sequoia.
https://github.com/user-attachments/assets/e3057220-308c-4a87-9b70-a6a3cd76ba71

**Comment by raoulkent** [comment]
That's odd, I've installed latest and it works fine, super snappy again
(Yabai 7.1.4, Sequoia 15.0 on M1 Pro)

**Comment by eret9616** [comment]
what I still find the problem is autofocus still fail after some operations.. but i haven't figure out how to reproduce it ..

**Comment by koekeishiya** [comment]
> I just updated to v7.1.4, and this is still a problem for me on Sequoia. https://github.com/user-attachments/assets/e3057220-308c-4a87-9b70-a6a3cd76ba71

I tested it on my 15.0 instance and it worked fine in v7.1.4 (v7.1.3 had the issue) and other people have also reported the new version fixing the issue for them, so realistically I am going to need something more than "still a problem" if you want me to be able to do something here.

**Comment by matzegebbe** [comment]
for me it also works fine now. Thank you @koekeishiya

**Comment by johnnyBira** [comment]
After my comment I tried it with other apps, and it works with all of them, except for Wezterm as demonstrated in my comment video attachment.
Before it was a problem with all apps, so I'm not sure what's different about Wezterm. Switching  between workspaces refreshes the layout and makes the windows (even Wezterm) behave as they should.

Thank you for all your effort, the last release was a great improvement.
